### PR TITLE
signal-desktop: init at 1.0.35

### PR DESCRIPTION
**Pull request by ixmatus**
###### Motivation for this change
The Signal Desktop application was recently released and the community has no derivation for it.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/signal-desktop/default.nix [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+      mv $out/usr/share $out/share
+      mv $out/opt/Signal $out/libexec
+      rmdir $out/usr $out/opt
+
+      chmod -R g-w $out
+
+      # Patch signal
+      patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" $out/libexec/signal-desktop
+      patchelf --set-rpath ${rpath}:$out/libexec $out/libexec/signal-desktop
```
Comment (NickHu) [inline pkgs/applications/networking/instant-messengers/signal-desktop/default.nix]: Perhaps better to just use one `patchelf` invocation

File: pkgs/applications/networking/instant-messengers/signal-desktop/default.nix [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    xlibs.libXext
+    xlibs.libXfixes
+    xlibs.libXi
+    xlibs.libXrandr
+    xlibs.libXrender
+    xlibs.libXtst
+    xlibs.libxcb
+    zulu
+    zulu8
+  ] + ":${stdenv.cc.cc.lib}/lib64";
```
Comment (NickHu) [inline pkgs/applications/networking/instant-messengers/signal-desktop/default.nix]: Quite a few unnecessary imports here, see https://github.com/NixOS/nixpkgs/pull/31337 for a minimal set

File: pkgs/applications/networking/instant-messengers/signal-desktop/default.nix [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+          sha256 = "d9f9d4d54f4121efc8eadf1cf0ff957828088b313e53b66dc540b851c44c1860";
+        }
+      else
+        throw "Signal for Desktop is not currently supported on ${stdenv.system}";
+
+    nativeBuildInputs = [ dpkg ];
+    unpackPhase = "true";
+    buildCommand = ''
+      mkdir -p $out
+      dpkg -x $src $out
```
Comment (NickHu) [inline pkgs/applications/networking/instant-messengers/signal-desktop/default.nix]: I think this should be in the `unpackPhase`

File: pkgs/applications/networking/instant-messengers/signal-desktop/default.nix [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      mkdir -p $out/bin
+      ln -s $out/libexec/signal-desktop $out/bin/signal-desktop
+
+      # Fix the desktop link
+      substituteInPlace $out/share/applications/signal-desktop.desktop \
+        --replace /opt/Signal/signal-desktop $out/bin/signal-desktop
+    '';
+
+    meta = {
+      description = "Signal messenger for the desktop.";
```
Comment (NickHu) [inline pkgs/applications/networking/instant-messengers/signal-desktop/default.nix]: I think the branding is Signal Desktop or something like that. Better to ask someone who knows?

**Review by NickHu — CHANGES_REQUESTED** [review]
It's a shame we both decided to package this on the same day, within 2 hours of each other...

File: pkgs/applications/networking/instant-messengers/signal-desktop/default.nix (hunk 3 above)
Comment (ixmatus) [reply pkgs/applications/networking/instant-messengers/signal-desktop/default.nix]: Yup, good point.

File: pkgs/applications/networking/instant-messengers/signal-desktop/default.nix (hunk 1 above)
Comment (ixmatus) [reply pkgs/applications/networking/instant-messengers/signal-desktop/default.nix]: Good point.

File: pkgs/applications/networking/instant-messengers/signal-desktop/default.nix (hunk 4 above)
Comment (ixmatus) [reply pkgs/applications/networking/instant-messengers/signal-desktop/default.nix]: I used what you had for the description in your PR.

**Comment by ixmatus** [comment]
Thank you for the review @NickHu!

**Comment by NeQuissimus** [comment]
Thanks guys!

**Pull request merged**